### ISSUE TITLE
Assassin holoparas stay in stealth mode when attacked or attacking

### ISSUE THREAD

**Issue by TheSardele** [ParadiseSS13/Paradise#12225]
**Issue Description**:
As in the title.

**What did you expect to happen**: 
For the guardian to have one powerful attack then become visible and less dangerous, as their description implies.

**What happened instead**:
The guardian dealt stealth damage with every hit and didn't turn visible.

**Why is this bad/What are the consequences**:
Having a guardian that is very hard to see and keep track of that deals 50 damage with every attack is ludicrously powerful.

**Steps to reproduce the problem**:
Get an assassin guardian, make all of security hit the floor before they realise what's going on.

**When did the problem start happening**:
Assassin guardians are rarely used so it's hard to tell, no clue.

**Extra information**:
When testing this I simply spawned in a holopara with admin powers, but I did so because I saw it happen during a live round.

**Comment by AzuleUtama** [comment]
I've tested this myself and it seems as though they are working correctly. Punching a mob while in stealth will throw you out of it and you'll deal normal damage again.

I tested this first by using a holoparasite injector, making a ghost into a holoparasite with it and then testing stealth without any issues. I then repeated this for the wizard variant.

Were they attacking anything specific during the incident? Alternatively it might have been fixed with #12157 depending on when this occurred.
